### PR TITLE
Refactor/league stats helper methods

### PR DESCRIPTION
**Pull request by lanebret85**
## Description

Refactored helper methods and tests for helper methods that correspond to 6 "offense" methods from league stats in team_result_parser.rb.

Edit: Also added code and tests for tackles methods to both parser classes and stat_tracker.

## Type of change

- [ ] fix
- [x] feat
- [x] test
- [x] refactor
- [ ] docs

## Checklist

- [x] code has been self reviewed
- [x] code runs without any errors
- [x] thorough testing has been implemented if adding feature
- [x] all tests pass

### Thanks!

### CODE REVIEW

**Review by aevans27 — APPROVED** [review]
Looks great, built the tackle methods with you.  Glad tests work!

File: lib/team_result_parser.rb [hunk 1]
```diff
@@ -152,53 +163,81 @@ def worst_offense
...
-  
-  def games_played_per_team #helper method
+
+  def find_best_worst_offense_team_id_by_location(location, goodness)
+    a = alltime_goals_per_team_by_location(location)
+    b = games_played_per_team_by_location(location)
+    
+    c = a.merge(b) { |team_id, goals, games| goals.to_f / games.to_f }
+    
+    if goodness == "best"
```
Comment (TLamb32) [inline lib/team_result_parser.rb:179]: Ok this is an awesome way to refactor this into one method.

File: lib/team_result_parser.rb [hunk 2]
```diff
@@ -234,102 +273,10 @@ def lowest_scoring_visitor
...
-        if game.team_id.to_i == counter
-          home_games_played[game.team_id.to_i] += 1
-        end
-      end
-      counter += 1
-    end
-    home_games_played
-  end
-
-  def visiting_games_played_per_team #helper method
```
Comment (TLamb32) [inline lib/team_result_parser.rb:293]: Sad to see this code go, but it was its time hah

**Review by TLamb32 — APPROVED** [review]
Great refactors. The amount of lines deleted is impressive.

**Pull request merged**